sw/device/tests/dif/dif_uart_sanitytest.c

```diff
@@ -0,0 +1,61 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_uart.h"
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/runtime/hart.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+static dif_uart_t uart0;
+static const uint8_t kSendData[] = "Sanity test!";
+
+static void uart_initialise(mmio_region_t base_addr, dif_uart_t *uart) {
+  dif_uart_config_t config = {
+      .baudrate = kUartBaudrate,
+      .clk_freq_hz = kClockFreqHz,
+      .parity_enable = kDifUartDisable,
+      .parity = kDifUartParityEven,
+  };
+
+  if (dif_uart_init(base_addr, &config, uart) != kDifUartConfigOk) {
+    abort();
```

Comment:
I guess the point of this test is to validate that send/receive functionality works before printing anything out. I think instead of using `test_status_set(kTestStatusFailed)` which uses `LOG_INFO`, I can avoid `abort` by constructing my test to return `false` to `main.c` in case of failure.

Thinking about it, I should also test non polled send/receive in this test.

Reply:
> Thinking about it, I should also test non polled send/receive in this test.

No, the test doesn't need to be comprehensive, it needs to test the main functionality (which you are doing), and execute reasonably quickly.

Reply:
`return false;` also works.